src/heitech.zer0mqXt.core/Transport/Message.cs

```diff
@@ -96,6 +99,18 @@ internal static XtResult<TMessage> ParseRqRepMessage<TMessage>(this NetMQMessage
             }
         }
 
-        public static string TypeFrameName(this Type type) => type.FullName;
+        ///<summary>
+        /// Get the TypeFrame by Convention (Type-Name) or by declarative MessageAttribute
+        ///</summary>
+        public static string TypeFrameName(this Type type) 
+        {
+            // todo optimize with static cache or others?
```

Comment:
Perhaps we could also have a source generator for that 🤔.

Reply:
yeah but it is optimization so I would not think about it at this point. And certainly not the implementation yet :D but that might be a viable sln.